lib/core/filtering/simple_rule.rb

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+module Core
+  module Filtering
+    class SimpleRule
+      POLICY = 'policy'
+      FIELD = 'field'
+      RULE = 'rule'
+      VALUE = 'value'
+      ID = 'id'
+
+      DEFAULT_RULE_ID = 'DEFAULT'
+      DEFAULT_RULE = SimpleRule.new(
+        {
+          'policy' => 'include',
+          'field' => '_',
+          'rule' => 'regex',
+          'value' => '.*',
+          'id' => SimpleRule::DEFAULT_RULE_ID
+        }
+      )
+
+      attr_reader :policy, :field, :rule, :value, :id
+
+      def initialize(rule_hash)
+        @policy = rule_hash.fetch(POLICY)
+        @field = rule_hash.fetch(FIELD)
+        @rule = rule_hash.fetch(RULE)
+        @value = rule_hash.fetch(VALUE)
+        @id = rule_hash.fetch(ID)
+      end
+
+      def match?(document)
+        return true if id == DEFAULT_RULE_ID
+        doc_value = document[field]
+        case rule
+        when 'equals'
+          doc_value == value
+        when 'starts_with'
+          doc_value.starts_with?(value)
+        when 'ends_with'
+          doc_value.ends_with?(value)
+        when 'contains'
+          doc_value.include?(value)
+        when 'regex'
```

Comment:
Actually, I thought that only `starts_with` and `ends_with` were to be covered in post-processing (by the way, why not in pre-processing?). Now I'm not sure what we have in pre-processing and what in post-processing?

Reply:
You're right, we could do this with regex for mongo. I didn't see an explicit "starts with" operator in their docs so I just left it, but maybe we should support that. Though I'm also fine to not scope creep that at this point.